File: skills/cow-arms-technology-analysis/scripts/python/03_cross_national.py
#!/usr/bin/env python3
"""03_cross_national.py — snapshot-year cross-national dot plot of total_use.

Sorts all states by total_use in the chosen year. Lollipop / dot-line is
more legible than bars for ranked data and prints cleanly in B&W.
"""

import os
from pathlib import Path
import pandas as pd
import matplotlib.pyplot as plt

SNAPSHOT_YEAR = 1950  # change to inspect another year

data_dir = Path(os.environ.get(
    "COW_ARMS_DATA_DIR",
    Path(__file__).resolve().parents[3] / "ArmsTechnologyV1",
))

wide = pd.read_csv(data_dir / "cow_arms_tech_wide.csv")
snap = (
    wide.loc[(wide.year == SNAPSHOT_YEAR) & wide.total_use.notna(),
             ["statename", "total_use"]]
    .sort_values("total_use")
    .reset_index(drop=True)
)

plt.rcParams.update({"font.family": "Helvetica", "font.size": 8})

fig, ax = plt.subplots(figsize=(7.2, 9.0))
y_pos = range(len(snap))
ax.hlines(y_pos, 0, snap["total_use"], color="grey", linewidth=0.3)
ax.plot(snap["total_use"], y_pos, "o", color="#0072B2", markersize=3.5)

ax.set(
    yticks=y_pos,
    yticklabels=snap["statename"].tolist(),
    xlim=(0, 29),
    xticks=range(0, 30, 5),
    xlabel=f"total_use in {SNAPSHOT_YEAR} (0–29)",
)
ax.tick_params(axis="y", labelsize=6)
ax.spines[["top", "right"]].set_visible(False)

fig.text(0.01, 0.005,
         "Source: COW Arms Technology Data Set v1.0 (Hariri & Wingender 2025).",
         fontsize=8, style="italic")
fig.tight_layout(rect=(0, 0.015, 1, 1))

Path("output").mkdir(exist_ok=True)
out = f"output/figure_crossnational_{SNAPSHOT_YEAR}.pdf"
fig.savefig(out, bbox_inches="tight")
print(f"Wrote {out} — {len(snap)} states")

File: skills/cow-arms-technology-analysis/scripts/R/03_cross_national.R
#!/usr/bin/env Rscript
# 03_cross_national.R — snapshot-year cross-national comparison.
#
# Plots total_use in a chosen year for all states in the international system,
# sorted by adoption level. Uses a dot plot (cleaner than bars for ranked data).

suppressPackageStartupMessages({
  library(readr)
  library(dplyr)
  library(ggplot2)
})

SNAPSHOT_YEAR <- 1950  # change to inspect another year

data_dir <- Sys.getenv("COW_ARMS_DATA_DIR",
                       unset = file.path("..", "..", "..", "..", "ArmsTechnologyV1"))

snap <- read_csv(file.path(data_dir, "cow_arms_tech_wide.csv"),
                 show_col_types = FALSE) |>
  filter(year == SNAPSHOT_YEAR, !is.na(total_use)) |>
  arrange(total_use) |>
  mutate(statename = factor(statename, levels = statename))

p <- ggplot(snap, aes(total_use, statename)) +
  geom_point(size = 1.4, color = "#0072B2") +
  geom_segment(aes(x = 0, xend = total_use, y = statename, yend = statename),
               color = "grey70", linewidth = 0.3) +
  scale_x_continuous(limits = c(0, 29), breaks = seq(0, 29, 5)) +
  labs(x = paste0("total_use in ", SNAPSHOT_YEAR, " (0–29)"),
       y = NULL,
       caption = "Source: COW Arms Technology Data Set v1.0 (Hariri & Wingender 2025).") +
  theme_minimal(base_size = 8) +
  theme(panel.grid.minor = element_blank(),
        axis.text.y = element_text(size = 6),
        plot.caption = element_text(size = 8, face = "italic", hjust = 0))

dir.create("output", showWarnings = FALSE)
ggsave(sprintf("output/figure_crossnational_%d.pdf", SNAPSHOT_YEAR), p,
       width = 7.2, height = 9.0, device = cairo_pdf)
cat(sprintf("Wrote output/figure_crossnational_%d.pdf — %d states\n",
            SNAPSHOT_YEAR, nrow(snap)))
